dbt/include/trino/macros/materializations/table.sql

```diff
@@ -65,9 +68,12 @@
       {%- endcall %}
   {% endif %}
 
-  {{ run_hooks(post_hooks) }}
-
   {% do persist_docs(target_relation, model) %}
 
+  {% set should_revoke = should_revoke(existing_relation, full_refresh_mode=True) %}
+  {% do apply_grants(target_relation, grant_config, should_revoke=should_revoke) %}
+
+  {{ run_hooks(post_hooks) }}
```

Comment:
Should it also be applicable to `incremental.sql`?

Reply:
Checked on dbt-snowflake and applied the same onto ours.

It seems not tested within core dbt. It's however the same hooks but only applied on full_refresh.

Views is tested and is working, although no hooks in the materialization are called.